main.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"os"
+
+	"github.com/google/go-github/v37/github"
+	"github.com/gorilla/mux"
+	"golang.org/x/oauth2"
+)
+
+//define global github client scoped to main package
+var client *github.Client
+
+func init() {
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: os.Getenv("ACCESS_KEY")},
+	)
+	tc := oauth2.NewClient(context.Background(), ts)
+	client = github.NewClient(tc)
+}
+
+func ProtectRepoHandler(w http.ResponseWriter, r *http.Request) {
+
+	req := github.WebHookPayload{}
```

Comment:
Instead of creating a model of the event payload from GHE, I decided to utilize an open source library (e.g., the Google Go GitHub API Client) to decode my request data into the appropriate model. Furthermore, this module also allows for creating requests to the GitHub API.